src/model/table/table-filter-types.ts

```diff
@@ -72,9 +73,15 @@ export class FilterFunctions {
       (total, filter) =>
         isNullOrUndefined(filter.operator) || filter.operator === '&&'
           ? total &&
-            FilterFunctions.getFilterFunction(filter.function.functionName)(row[filter.columnAccessor], filter.value)
+            FilterFunctions.getFilterFunction(filter.function.functionName)(
+              get(row, filter.columnAccessor),
```

Comment:
We've talked about it on our meeting and I explained logic behind those functions in a nutshell.

![image](https://user-images.githubusercontent.com/48626088/121069293-995a7f00-c7cd-11eb-9bfa-c20e8f5e30aa.png)

1/ I need to check if operator (`&&` or `||`) exist to create union or intersection between filters.
2/ Second I refer to static methods by names (like `isNotEmpty`, `contains` etc.) - that's what `getFilterFunction` does.
3/ At the end I just pass values to filter function - I use `get` from `lodash` to access deeply nested properties like `category.tax.value`.
4/ Everything is compounded using `reduce` function to return simple `boolean` type as the result.

If you want to understand it completely, you would probably have to write it for your own or sit on it for more time. 😃

Reply:
I have never heard about rambda.js

I checked that vulnerabilities that you've mentioned.
I uninstalled `lodash` completely and they stayed exactly the same - 199 moderate and 3 high, nothing changed.
That's because a lot of libraries that we have in the project are actually using `lodash` under the hood - even `chakra`.
![image](https://user-images.githubusercontent.com/48626088/121069635-ff470680-c7cd-11eb-9f97-9068126b2426.png)
![image](https://user-images.githubusercontent.com/48626088/121069662-0706ab00-c7ce-11eb-881c-0ee372fbc1ee.png)

I prefer to use `lodash`, because I know that library quite a bit and it has nice set of helpful functions. In addition, we are already using it under the hood (we can say that).

FYI - I have to add it into `package.json` because compiler gives error about using some kind of "_shadow import/package_". 
I don't remember the error exactly right now, but I need to import and install it manually also.

Said that, I would prefer to stay this way.